receiver/dotnetdiagnosticsreceiver/receiver_test.go

```diff
@@ -27,20 +28,81 @@ import (
 
 func TestReceiver(t *testing.T) {
 	ctx := context.Background()
+	rw := &fakeRW{writeErrOn: -1}
 	r, err := NewReceiver(
 		ctx,
-		zap.NewNop(),
-		createDefaultConfig().(*Config),
 		consumertest.NewMetricsNop(),
-		fakeConnect,
+		func() (io.ReadWriter, error) {
+			return rw, nil
+		},
+		nil,
+		1,
+		zap.NewNop(),
 	)
 	require.NoError(t, err)
 	err = r.Start(ctx, componenttest.NewNopHost())
 	require.NoError(t, err)
 	err = r.Shutdown(ctx)
 	require.NoError(t, err)
+	const magic = "DOTNET_IPC_V1"
+	magicBytes := rw.writeBuf[:len(magic)]
+	require.Equal(t, magic, string(magicBytes))
+}
+
+func TestReceiver_ConnectError(t *testing.T) {
+	ctx := context.Background()
+	r, err := NewReceiver(
+		ctx,
+		consumertest.NewMetricsNop(),
+		func() (io.ReadWriter, error) {
+			return nil, errors.New("")
+		},
+		nil,
+		1,
+		zap.NewNop(),
+	)
+	require.NoError(t, err)
+	err = r.Start(ctx, componenttest.NewNopHost())
+	require.Error(t, err)
+}
+
+func TestReceiver_WriteRequestError(t *testing.T) {
+	ctx := context.Background()
+	rw := &fakeRW{}
+	r, err := NewReceiver(
+		ctx,
+		consumertest.NewMetricsNop(),
+		func() (io.ReadWriter, error) {
+			return rw, nil
+		},
+		nil,
+		1,
+		zap.NewNop(),
+	)
+	require.NoError(t, err)
+	err = r.Start(ctx, componenttest.NewNopHost())
+	require.Error(t, err)
```

Comment:
same as https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/2365/files#r580487442

Reply:
Thanks, I factored out the common code.